### DISCUSSION TITLE
Calendar View

### DISCUSSION THREAD

**Discussion by srid** [srid/emanote#183] [Ideas]
Writing daily notes is a Zettelkasten pattern. Emanote should render a sidebar calendar similar to this:

![image](https://user-images.githubusercontent.com/3998/139552149-cc1a10f8-bec4-4021-a5be-e342f2531d33.png)

.. linking to the various daily notes. The image is from https://reflect.academy/

---

**Tangential idea**: Can we implement an actual event calendar too? ie., if `Foo.md` is marked as "repeats every Saturday", then it would appear on Saturday's in the calendar.

**Comment by j-steinbach** [comment]
> ie., if Foo.md is marked as "repeats every Saturday", then it would appear on Saturday's in the calendar.

How would you mark that? In the text? As front-matter?

**Reply by srid** [reply]
Not sure yet.

A naive approach would go for using frontmatter. Doing it in text means we likely need *labelled* wiki-links, where the 'label' encodes the 'repeats ever' information, with the link encoding date/day. Or something like that.

I suppose frontmatter is better:

```yaml
calendar:
  - childOf: "every Saturday"  # or "2021-02-23" (which is same as `#[[2021-02-23]]` in text)
```

... assuming it has Emanote create those folgezettel relations implicitly, to all Saturday daily notes. So the rendering/view part don't have to deal with it, and can just continue to show folgezettel children as the day's notes.

**Reply by j-steinbach** [reply]
I think I also prefer the front-matter approach, but I am not sure I understand how/why I want to link my daily notes -- or more specifically: what do you mean with "repeat"?. What exactly do you mean by that?

If I have a directory of daily notes (or notes tagged as "daily), and they are all named after a specific date (`2021-10-31`), why I can't I just point the calendar to that directory and render them?

---

Asides from that it would make sense if "daily notes" would be intrinsically linked. I.e. they automatically (based on their tag/filename/folder)  link to the previous and next day. (See https://github.com/srid/emanote/discussions/147 for ideas)

**Reply by srid** [reply]
Yes, implicit in the idea is to automatically place notes based on their filename (ie., `2021-02-21.md` or `2022.md` or `2021-04.md`) in their respective entry in the calendar. This works for 1-on-1 placement, but not for say 1-on-many placements, hence the YAML frontmatter syntax above - which effectively places that one note under many days (Saturday's) in the calendar.

This example https://inspirativity.zettel.page - should "just work" with the calendar view, without any specific change.

**Reply by srid** [reply]
> I am not sure I understand how/why I want to link my daily notes 

Another case for 1-on-many placement is to place permanent notes manually under a certain date. You can do this as:

```markdown

# My blog post

...

This post was filed under day #[[2021-03-24]]
```

And the calendar view will show it automatically under that day. It would effectively be the same as adding a `date` field to YAML.

> or more specifically: what do you mean with "repeat"?. What exactly do you mean by that?

The specific use-case for the "repeat" idea is to allow me to review a particular note periodically. So it will appear automatically, say, every weekend as I get to writing the then daily note. I thought I'd extend this idea more generally to calendar events; eg: a note for Monday work meetings (each meeting may have its own meeting-notes as a folgezettel descendent of this note).

**Reply by j-steinbach** [reply]
*(Don't have much time to answer in-depth, here are some scattered impressions)*



It would make sense to combine this review feature with a push-notification. I enter the Emanote site and on top I get some "notification cards" that link me to a "repeat" note.

---

Emanote could/should also display the last-viewed timestamp on any note I visited. This could be combined with the discovery.

---

Having notes on repeat very much reminds me of [Anki](https://apps.ankiweb.net/).
